ansible_base/lib/utils/views.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import time
+
+from django.conf import settings
+from django.db import connection
+from django.utils.translation import gettext_lazy as _
+from rest_framework import views
+from rest_framework.exceptions import AuthenticationFailed, ParseError, PermissionDenied, UnsupportedMediaType
+
+from ansible_base.lib.utils.settings import get_function_from_setting
+
+logger = logging.getLogger('ansible_base.lib.utils.views')
+
+
+class AnsibleBaseView(views.APIView):
+    def initialize_request(self, request, *args, **kwargs):
+        """
+        Store the Django REST Framework Request object as an attribute on the
+        normal Django request, store time the request started.
+        """
+        self.time_started = time.time()
+        if getattr(settings, 'SQL_DEBUG', False):
+            self.queries_before = len(connection.queries)
+
+        # If there are any custom headers in REMOTE_HOST_HEADERS, make sure
+        # they respect the allowed proxy list
+        if all(
+            [
+                settings.PROXY_IP_ALLOWED_LIST,
+                request.environ.get('REMOTE_ADDR') not in settings.PROXY_IP_ALLOWED_LIST,
+                request.environ.get('REMOTE_HOST') not in settings.PROXY_IP_ALLOWED_LIST,
+            ]
+        ):
+            for custom_header in settings.REMOTE_HOST_HEADERS:
+                if custom_header.startswith('HTTP_'):
+                    request.environ.pop(custom_header, None)
+
+        drf_request = super().initialize_request(request, *args, **kwargs)
+        request.drf_request = drf_request
+        try:
+            request.drf_request_user = getattr(drf_request, 'user', False)
```

Comment:
I this this exists to pass data to a method `get_current_user_from_drf_request` which is integrated deeply with the CRUM library. The functionality is working without it so hopefully we don't need it.

Reply:
Removed.